File: CardMasterSkin/Skins/SkinsProject.vb
Imports System.IO
Imports Newtonsoft.Json

Namespace Skins

    Public Class SkinsProject

        Public Property TexturesDirectory As String
        Public Property ImagesDirectory As String

        Public Shared Function LoadProject(file As FileInfo) As SkinsProject

            Dim skinsProject As SkinsProject = Nothing

            If file IsNot Nothing AndAlso file.Exists Then

                Dim sr As New StreamReader(file.FullName)
                Dim js As String = sr.ReadToEnd

                skinsProject = JsonConvert.DeserializeObject(Of SkinsProject)(js)

                sr.Close()
                sr.Dispose()

                skinsProject.TexturesDirectory = skinsProject.TexturesDirectory.Replace("\\", "\")
                skinsProject.ImagesDirectory = skinsProject.ImagesDirectory.Replace("\\", "\")

            End If

            Return skinsProject

        End Function

        Public Sub Save(file As FileInfo)

            file.Delete()

            Dim js As String = JsonConvert.SerializeObject(Me, Formatting.Indented)
            Dim sw As New StreamWriter(file.FullName, FileMode.OpenOrCreate)

            sw.Write(js)

            sw.Close()
            sw.Dispose()

        End Sub

    End Class

End Namespace

File: CardMasterSkin/Skins/SkinsProject.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.IO;

namespace CardMasterSkin.Skins
{
    public class SkinsProject
    {
        public Dictionary<String, String> MapLibelleColor { get; set; }
        public Dictionary<String, String> MapKindField { get; set; }
        public Dictionary<String, String> MapLibelleBorderColor { get; set; }
        public Dictionary<String, String> MapRareteColor { get; set; }
        public int BorderWidth { get; set; }
        public List<JsonSkin> Skins { get; set; }

        public static SkinsProject LoadProject(FileInfo file)
        {
            SkinsProject skinsProject = null;

            if ((file != null) && (file.Exists))
            {
                var sr = new StreamReader(file.FullName);
                string js = sr.ReadToEnd();

                skinsProject = JsonConvert.DeserializeObject<SkinsProject>(js);

                sr.Close();
                sr.Dispose();
            }

            return skinsProject;

        }

        public void Save(FileInfo file)
        {
            file.Delete();

            string js = JsonConvert.SerializeObject(this, Formatting.Indented);
            var sw = new StreamWriter(file.FullName, false);

            sw.Write(js);

            sw.Close();
            sw.Dispose();

        }

    }
}
